common/src/service/AuthenticationMethodFactory.cs

```diff
@@ -15,7 +15,7 @@ static internal IAuthenticationMethod GetAuthenticationMethod(IotHubConnectionSt
         {
             if (!string.IsNullOrWhiteSpace(iotHubConnectionStringBuilder.DeviceId))
             {
-                return new ServiceAuthenticationWithDeviceCredentials(iotHubConnectionStringBuilder.DeviceId, iotHubConnectionStringBuilder.SharedAccessKey);
+                return new ServiceAuthenticationWithDeviceCredentials(iotHubConnectionStringBuilder.DeviceId, iotHubConnectionStringBuilder.SharedAccessKey, iotHubConnectionStringBuilder.SharedAccessSignature);
```

Comment:
This doesn't really make sense: you already have the Shared Access Key which is used to generate a signature (SAS). 
Why would we put both in this class ctor?